asm/instruction.go

```diff
@@ -441,9 +521,43 @@ func (insns Instructions) Size() uint64 {
 	return sum
 }
 
+// AssociateMap rewrites all loads of a specific map pointer.
+//
+// Returns an error if the symbol isn't used, see IsUnreferencedSymbol.
+func (insns Instructions) AssociateMap(symbol string, m FDer) error {
+	if symbol == "" {
+		return errors.New("empty symbol")
+	}
+
+	found := false
+	for i := range insns {
+		ins := &insns[i]
+		if ins.Reference() != symbol {
+			continue
+		}
+
+		if err := ins.AssociateMap(m); err != nil {
+			return err
+		}
+
+		// Remove the reference, since we have now resolved it, so we will not attempt to do so again later.
+		*ins = ins.WithReference("")
+
+		found = true
+	}
+
+	if !found {
+		return &unreferencedSymbolError{symbol}
```

Comment:
This (barely) predates `%w`, let's make this an unexported sentinel.

Reply:
No quite sure what you mean, do you want to replace this line with `fmt.Errorf("unreferenced symbol %s", symbol)` or do you wat to export `unreferencedSymbolError` and remove the `IsUnreferencedSymbol` so it can be used with the `errors.Is` function?